src/js/node/net.ts

```diff
@@ -161,1034 +168,1009 @@ function onConnectEnd() {
+        this.emit("_tlsError", error);
+        this.server.emit("tlsClientError", error, data);
+        SocketHandlers.error(socket, error, true);
+        return;
       }
+    }
+    SocketHandlers.error(socket, error, true);
+    data.server.emit("clientError", error, data);
+  },
+  timeout: SocketHandlers.timeout,
+  drain: SocketHandlers.drain,
+  binaryType: "buffer",
+};
 
-      if ($isArray(data)) {
-        // Was a writev, iterate over chunks to get total length
-        for (let i = 0; i < data.length; i++) {
-          const chunk = data[i];
-
-          if (data.allBuffers || chunk instanceof Buffer) bytes += chunk.length;
-          else bytes += Buffer.byteLength(chunk.chunk, chunk.encoding);
+function Socket(options?) {
+  const {
+    socket,
+    signal,
+    write,
+    read,
+    allowHalfOpen = false,
+    onread = null,
+    noDelay = false,
+    keepAlive = false,
+    keepAliveInitialDelay = 0,
+    ...opts
+  } = options || {};
+
+  if (options?.objectMode) throw $ERR_INVALID_ARG_VALUE("options.objectMode", options.objectMode, "is not supported");
+  if (options?.readableObjectMode)
+    throw $ERR_INVALID_ARG_VALUE("options.readableObjectMode", options.readableObjectMode, "is not supported");
+  if (options?.writableObjectMode)
+    throw $ERR_INVALID_ARG_VALUE("options.writableObjectMode", options.writableObjectMode, "is not supported");
+
+  Duplex.$call(this, {
+    ...opts,
+    allowHalfOpen,
+    readable: true,
+    writable: true,
+    //For node.js compat do not emit close on destroy.
+    emitClose: false,
+    autoDestroy: true,
+    // Handle strings directly.
+    decodeStrings: false,
+  });
+  this._parent = this;
+  this._parentWrap = this;
+  this[kpendingRead] = undefined;
+  this[kupgraded] = null;
+
+  this[kSetNoDelay] = Boolean(noDelay);
+  this[kSetKeepAlive] = Boolean(keepAlive);
+  this[kSetKeepAliveInitialDelay] = ~~(keepAliveInitialDelay / 1000);
+
+  this[khandlers] = SocketHandlers;
+  this.bytesRead = 0;
+  this[kBytesWritten] = undefined;
+  this[kclosed] = false;
+  this[kended] = false;
+  this.connecting = false;
+  this.localAddress = "127.0.0.1";
+  this.remotePort = undefined;
+  this[bunTLSConnectOptions] = null;
+  this.timeout = 0;
+  this[kwriteCallback] = undefined;
+  this._pendingData = undefined;
+  this._pendingEncoding = undefined; // for compatibility
+  this[kpendingRead] = undefined;
+  this._hadError = false;
+  this.isServer = false;
+  this._handle = null;
+  this._parent = undefined;
+  this._parentWrap = undefined;
+  this[ksocket] = undefined;
+  this.server = undefined;
+  this.pauseOnConnect = false;
+  this[kupgraded] = undefined;
+
+  // Shut down the socket when we're finished with it.
+  this.on("end", onSocketEnd);
+
+  if (socket instanceof Socket) {
+    this[ksocket] = socket;
+  }
+  if (onread) {
+    if (typeof onread !== "object") {
+      throw new TypeError("onread must be an object");
+    }
+    if (typeof onread.callback !== "function") {
```

Comment:
should it be validateFunction

Reply:
node doesnt throw this error it seems so ill audit this again once this lands